File: Examples/Getting-Started/Get-Job-and-Run-Now/nodeJS/get-job-and-run-now.mjs
import "dotenv/config";

import { NetLDClient, NetLDError, printJson } from "./netld-example-client.mjs";

const jobName = process.env.NETLD_JOB_NAME;
if (!jobName) {
  throw new NetLDError("Set NETLD_JOB_NAME to the exact name of the job to run.");
}

const client = NetLDClient.fromEnv();
await client.login();

const pageData = await client.searchAvailableJobs({
  networks: [process.env.NETLD_NETWORK || "Default"],
  pageSize: Number.parseInt(process.env.NETLD_JOB_PAGE_SIZE || "100", 10),
});

const matches = (pageData?.jobData || []).filter((job) => job.jobName === jobName);

if (matches.length === 0) {
  throw new NetLDError(`No available job named "${jobName}" was found.`);
}
if (matches.length > 1) {
  throw new NetLDError(
    `Multiple jobs named "${jobName}" were found: ${matches.map((job) => job.jobId).join(", ")}`,
  );
}

const jobId = matches[0].jobId;
const jobData = await client.getJob(jobId);
if (!jobData) {
  throw new NetLDError(`Scheduler.getJob returned no data for job ID ${jobId}.`);
}

console.log(`Selected job "${jobName}" with ID ${jobId}:`);
printJson(jobData);

if ((process.env.NETLD_RUN_JOB || "false").toLowerCase() !== "true") {
  console.log("Dry run only. Set NETLD_RUN_JOB=true to execute this job.");
} else {
  const execution = await client.runNow(jobData);
  console.log("Execution started:");
  printJson(execution);
}

File: Examples/Getting-Started/Get-Job-and-Run-Now/Python/get_job_and_run_now.py
import os

from netld_example_client import NetLDClient, NetLDError, print_json


job_name = os.environ.get("NETLD_JOB_NAME")
if not job_name:
    raise NetLDError("Set NETLD_JOB_NAME to the exact name of the job to run.")

client = NetLDClient.from_env()
client.login()

page_data = client.search_available_jobs(
    networks=[os.environ.get("NETLD_NETWORK", "Default")],
    page_size=int(os.environ.get("NETLD_JOB_PAGE_SIZE", "100")),
)

matches = [
    job
    for job in (page_data or {}).get("jobData", [])
    if job.get("jobName") == job_name
]

if not matches:
    raise NetLDError(f'No available job named "{job_name}" was found.')
if len(matches) > 1:
    job_ids = ", ".join(str(job.get("jobId")) for job in matches)
    raise NetLDError(f'Multiple jobs named "{job_name}" were found: {job_ids}')

job_id = matches[0]["jobId"]
job_data = client.get_job(job_id)
if job_data is None:
    raise NetLDError(f"Scheduler.getJob returned no data for job ID {job_id}.")

print(f'Selected job "{job_name}" with ID {job_id}:')
print_json(job_data)

if os.environ.get("NETLD_RUN_JOB", "false").lower() != "true":
    print("Dry run only. Set NETLD_RUN_JOB=true to execute this job.")
else:
    execution = client.run_now(job_data)
    print("Execution started:")
    print_json(execution)
